test/flush_worker_test.cc

```diff
@@ -0,0 +1,130 @@
+#include <functional>
+#include <thread>
+
+#include "external/envoy/source/common/common/random_generator.h"
+#include "external/envoy/source/common/event/dispatcher_impl.h"
+#include "external/envoy/source/common/runtime/runtime_impl.h"
+#include "external/envoy/source/common/stats/isolated_store_impl.h"
+#include "external/envoy/test/mocks/api/mocks.h"
+#include "external/envoy/test/mocks/local_info/mocks.h"
+#include "external/envoy/test/mocks/protobuf/mocks.h"
+#include "external/envoy/test/mocks/stats/mocks.h"
+#include "external/envoy/test/mocks/thread_local/mocks.h"
+#include "external/envoy/test/test_common/simulated_time_system.h"
+
+#include "client/flush_worker_impl.h"
+
+#include "gtest/gtest.h"
+
+namespace Nighthawk {
+namespace Client {
+
+using namespace testing;
+
+class FlushWorkerTest : public Test {
+public:
+  FlushWorkerTest() : thread_factory_(Envoy::Thread::threadFactoryForTest()) {
+    dispatcher_ = new NiceMock<Envoy::Event::MockDispatcher>();
+    loader_ = std::make_unique<Envoy::Runtime::ScopedLoaderSingleton>(
+        Envoy::Runtime::LoaderPtr{new Envoy::Runtime::LoaderImpl(
+            *dispatcher_, tls_, {}, local_info_, store_, rand_, validation_visitor_, mock_api_)});
+    sink_ = new StrictMock<Envoy::Stats::MockSink>();
```

Comment:
Maybe remove this field, and add a getter to obtain this pointer from the `stats_sinks_` vector?

Reply:
Removing sink_ and replace it with the pointer getting from the stats_sinks_ seems not work in `EXPECT_CALL(*sink_, flush(_))` since stats_sinks_ is a list of `Sink` not `MockSink`

Reply:
Would there be anything against making stats_sinks_ a list of `MockSink`?

Reply:
If stats_sinks_ changes to a list of MockSink, then we got error `no matching constructor for initialization of 'Nighthawk::Client::FlushWorkerImpl'` when creating the FlushWorkerImpl 

`auto worker = std::make_unique<FlushWorkerImpl>(
      mock_api_, tls_, store_, stats_sinks_, stats_flush_interval_);`

Reply:
I see this makes sense then. Sorry about the noise